### DISCUSSION TITLE
p/2d9be16aabe14b50abff58cf979b9ec3/

### DISCUSSION THREAD

**Discussion by giscus** [rxliuli/blog#19] [Announcements]
# p/2d9be16aabe14b50abff58cf979b9ec3/

问题在编写一个重载函数时，吾辈发现了 ts 的方法签名问题。 12345678910111213141516171819enum TypeEnum {  A,  B,}type A = {  a: string}type B = {  b: number}//region 普通参数function fn1(type: TypeEnum.A, obj: A): voidfunction fn1(ty

https://blog.rxliuli.com/p/2d9be16aabe14b50abff58cf979b9ec3/

**Comment by leila-huang** [comment]
感谢分享，学习并应用到实际中了。感觉泛型和高阶的这么使用确实没想到T T，平时用的太浅层了

**Reply by rxliuli** [reply]
能帮到你就好，3 年前吾辈也在 ts 的复杂类型中挣扎，你也可以去看看 ts 官方文档：https://www.typescriptlang.org/docs/handbook/2/types-from-types.html

**Reply by leila-huang** [reply]
感谢感谢，赶紧去进阶一下了~~
